crux-mir/src/Mir/Intrinsics.hs

```diff
@@ -1709,3 +1932,104 @@ getSlicePtr e = getStruct i1of2 e
 
 getSliceLen :: Expr MIR s (MirSlice tp) -> Expr MIR s UsizeType
 getSliceLen e = getStruct i2of2 e
+
+
+--------------------------------------------------------------------------------
+-- ** MethodSpec and MethodSpecBuilder
```

Comment:
My hope is that someday we can eventually, generalize the stuff from here down, move it to the `crux` package, and use it for other languages, too.